src/components/Invoice.tsx

```diff
@@ -8,108 +8,171 @@ import Button from '@mui/material/Button';
 import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import IconButton from '@mui/material/IconButton';
+import { SubmitHandler, useForm, Controller } from 'react-hook-form';
+
+
+interface IPersonalData {
+
+  companyName: string | null;
+  city: string | null;
+  street: string | null;
+  postcode: string | null;
+  nip: number | null;
+  phoneNumber: number | null;
+  email: string | null,
+  bankAccount: string | null;
+}
+
+interface IInvoiceForm {
+  invoiceNumber: null,
+  dateFrom: Date | null;
+  dateTo: Date | null;
+  recipient: IPersonalData;
+  sender: IPersonalData
+}
 
 export default function Invoice() {
+  const { handleSubmit, register, control } = useForm<IInvoiceForm>({
+    defaultValues: {
+      invoiceNumber: null,
```

Comment:
The performance gain is really small here, but I would rather create a defaultValues const outside the component and only pass the reference here. It can also help with readability.